aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -95,7 +95,11 @@ Tensor& ceil_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(
 Tensor ceil(const Tensor& self) { return unary_op_impl(self, at::ceil_out); }
 Tensor& ceil_(Tensor& self) { return unary_op_impl_(self, at::ceil_out); }
 
-Tensor& expm1_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, expm1_stub); }
+Tensor& expm1_out(Tensor& result, const Tensor& self) { 
+  if (result.defined()) return unary_op_impl_out(result, self, expm1_stub, /*strategy_promote=*/ true);
```

Comment:
This enables dtype prmotion support for `out=...` arguments to the `expm1` function. Sample outputs:

```
input: torch.expm1(x /*dtype=torch.float16*/, out = torch.zeros((), dtype=torch.float32, device="cuda")).dtype
output: float32

input: torch.expm1(x /*dtype=torch.float16*/, out = torch.zeros((), dtype=torch.double, device="cuda")).dtype
output: float64
```

Reply:
`result.defined()` is always going to be true at this point, I think.